File: spec/python/test_switch_cast.py
import unittest

from switch_cast import SwitchCast

class TestSwitchCast(unittest.TestCase):
    def test_switch_cast(self):
        r = SwitchCast.from_file("src/switch_opcodes.bin")

        self.assertEqual(r.first_obj.value, "foobar")
        self.assertEqual(r.second_val, 0x42)
        # unable to test "err_cast" here

File: spec/lua/test_switch_cast.lua
local luaunit = require("luaunit")

require("switch_cast")

TestSwitchCast = {}

function TestSwitchCast:test_switch_cast()
    local r = SwitchCast:from_file("src/switch_opcodes.bin")

    luaunit.assertEquals(r.first_obj.value, "foobar")
    luaunit.assertEquals(r.second_val, 0x42)
    -- unable to test "err_cast" here
end
